R/input-formats.R

```diff
@@ -3,24 +3,53 @@
 #' In order to format validation docs, mrgvalidate must collect various pieces
 #' of information from the caller. This information can be broken down into two
 #' categories: information about the requirements and information about the
-#' executed tests. Below is a description of how information about the executed
-#' tests is expected to be formatted.
+#' executed tests. Below is a description of how this data is expected to be
+#' formatted. **Note: there a companion package `mrgvalprep` containing
+#' functions to preprocess common inputs (R's `testthat`, Googlesheets, etc.)
+#' into the format required by `mrgvalidate`.
 #'
 #' @details
 #'
+#' ## Stories and requirements
+#'
+#' A tibble or data.frame containing the stories, and optionally requirements,
+#' for the software being validated. Will be passed to `specs` argument of
+#' [create_validation_docs()]. Must have the following columns:
+#'
+#' * StoryId: unique identifier for the story
+#'
+#' * StoryName: short name of story (ideally 5-8 words max, but no hard limit)
+#'
+#' * StoryDescription: body of the story ("As a user,...")
+#'
+#' * ProductRisk: specify the risk to the product of this story (typically
+#' "Low", "Medium", "High", etc.)
+#'
+#' * RequirementId (Optional): unique identifier of technical requirement
+#' associated with story. If included, there can be >1 requirement per story,
+#' each on its own line, in which case the first four columns will be repeated
+#' for each requirement.
+#'
+#' * RequirementDescription (Optional): plain text description of the
+#' requirement
+#'
+#' * TestIds: unique identifiers for tests that validate this requirement or
+#' story. This field is used to map stories/requirements to tests.
+#'
 #' ## Automated tests
 #'
-#' For automated tests, a directory should contain set of CSVs with the test
-#' results. A CSV file must have the following columns:
+#' For automated tests, a directory containing any number of CSVs with the test
+#' results. The path will be passed to `auto_test_dir` argument of
+#' [create_validation_docs()]. A CSV file must have the following columns:
 #'
 #' * TestName: a description of the test
 #'
 #' * passed, failed: a count of passed and failed assertions for the test
 #'
 #' * TestId: a unique identifier of the test. This field is used to map tests
-#' to requirements.
+#' to stories/requirements.
 #'
-#' For `testthat` tests, the [parse_testthat_list_reporter()] helper can be used
+#' For `testthat` tests, the [mrgvalprep::parse_testthat_list_reporter()] helper can be used
```

Comment:
I _think_ this is going to give rcmdcheck fits. As in, since you're referencing the code, even in docs, it might try to link that information and thus want to enforce a dependent relationship. Something to keep an eye on.

Its either that, or not having that package present on the system will cause failure to build. I had to do some cleaning around that like 6 months ago cuz of docs like this so I'm hazy on it.

Reply:
you are correct. I changed it here https://github.com/metrumresearchgroup/mrgvalidate/commit/f28728247f7cbe3ceaf71451dc5b14e15fd5a10d

I guess this only works for packages you have in your `Imports`, which makes sense.